### PR TITLE
Amazon device messaging

### PR DESCRIPTION
**Pull request by darrylyip**
Hi, I added support for Amazon Device Messaging to work with Kindle devices.  Currently, ADM requires a POST request for each notification to be sent to a device.  They are working on functionality to support bulk notifications that can made with one request.  Once they roll that out, I will update accordingly.

I also added rapns:notifications:clean to allow completed rapns_notifications to be cleaned up.

Specs can be found in their respective adm directories. Please let me know if there is anything else I may have missed.  Thanks!

### CODE REVIEW

**Comment by ileitch** [comment]
Sorry for not merging this yet. I am currently on vacation/travelling. I will merge soon.

File: README.md [hunk 1]
```diff
@@ -78,6 +75,28 @@ n.save!
...
+app = Rapns::Adm::App.new
+app.name = "kindle_app"
+app.client_id = "..."
+app.client_secret = "..."
+app.connections = 1
+app.save!
+```
+
+```ruby
+n = Rapns::m::Notification.new
```
Comment (masterkain) [inline README.md:90]: typo here

**Comment by masterkain** [comment]
tested, seems to work fine. :+1:

**Comment by ileitch** [comment]
Can you please rebase this?

**Comment by kwasimensah** [comment]
Any ETA on when this will be wrapped into the next gem?

**Comment by ileitch** [comment]
Soon! I'm renaming rapns to rpush, keep and eye on https://github.com/rpush/rpush

**Comment by mattconnolly** [comment]
I like the name!

**Pull request merged**